### PR TITLE
BREAKING: Config#loadConfig returns Immutable.Map config object

### PR DESCRIPTION
**Pull request by lumberj**
config = require('lib/config').loadConfig('prefix') // prefix is the env var
prefix
config.getIn(['db', 'uri']) // returns config value at a given path
config.toJS() //returns the object representation of the config

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -1,150 +1,193 @@
 'use strict'
 
+const Immutable = require('immutable')
 const url = require('url')
 const tweetnacl = require('tweetnacl')
 
+function isRunningTests () {
+  return process.env.NODE_ENV.match(/unit|test/) &&
```
Comment (clark800) [inline lib/config.js]: The point of the lines below was to avoid having to set "NODE_ENV"

File: lib/config.js [hunk 2]
```diff
@@ -1,150 +1,193 @@
...
- * @example
- * class MyConfig extends Config {
- *   constructor () {
- *     super('example')
- *     this.parseServerConfig()
- *     this.parseDatabaseConfig()
- *   }
- * }
+ * Environment variables are passed in as strings, but this function can turn
+ * values like `undefined`, `''`, `'0'` and `'false'` into `false`.
```
Comment (clark800) [inline lib/config.js]: Should we take this opportunity to tighten this up? https://github.com/interledger/five-bells-ledger/commit/56e85c3e477c63333965eb17505d8e49d8cc6610#commitcomment-15468958

Comment (lumberj) [inline lib/config.js]: Can we save that for another PR? I'm used to using `1` and `0` as `true` and `false` so I see no reason to change it here.

File: lib/config.js (hunk 1 above)
Comment (lumberj) [inline lib/config.js]: I can remove it, but then testing configurations that are not test specific will be trickier

File: lib/config.js (hunk 2 above)
Comment (clark800) [inline lib/config.js]: ok

File: lib/config.js (hunk 1 above)
Comment (clark800) [inline lib/config.js]: You could use a new environment variable to force using the config settings even when running tests

File: lib/config.js [hunk 3]
```diff
@@ -1,150 +1,193 @@
...
+  const db = parseDatabaseConfig(prefix, localConfig && localConfig.db)
+  const keys = parseKeyConfig(prefix, localConfig && localConfig.keys)
+
+  const commonConfig = Immutable.fromJS({server, db, keys})
+
+  return commonConfig.merge(localConfig || {})
+}
 
+module.exports = {
+  getEnv,
```
Comment (clark800) [inline lib/config.js]: Can we make it only export `loadConfig` publicly?

Comment (clark800) [inline lib/config.js]: Nevermind, I guess we want these for generating the local config.

File: test/configSpec.js [hunk 4]
```diff
@@ -1,71 +1,292 @@
...
-      const config = new Config('a-long-PrEFiX')
+let originalEnv
+describe('config', () => {
+  beforeEach(() => {
+    originalEnv = _.cloneDeep(process.env)
+  })
 
-      expect(config.uppercasePrefix).to.equal('A_LONG_PREFIX_')
-    })
+  afterEach(() => {
```
Comment (clark800) [inline test/configSpec.js]: You can get rid of this if you just do `const originalEnv = _.cloneDeep(process.env)` at the top and `process.env = _.cloneDeep(originalEnv)` in `beforeEach`

File: lib/config.js [hunk 5]
```diff
@@ -1,150 +1,196 @@
...
 const url = require('url')
 const tweetnacl = require('tweetnacl')
 
+function isRunningTests () {
+  return process.argv[0].endsWith('mocha') ||
+    (process.argv.length > 1 && process.argv[0].endsWith('node') &&
+     process.argv[1].endsWith('mocha'))
+}
+
+function useTestCofig () {
```
Comment (clark800) [inline lib/config.js]: typo "cofig"

Comment (lumberj) [inline lib/config.js]: Fixed

File: test/configSpec.js (hunk 4 above)
Comment (lumberj) [inline test/configSpec.js]: Fixed

**Comment by clark800** [comment]
LGTM

**Comment by lumberj** [comment]
Holding off on merging as this is breaking. Working on the corresponding PRs to the other components

**Comment by MatthewPhinney** [comment]
LGTM

**Pull request merged**